### PR TITLE
created login/registration pages

### PR DESCRIPTION
**Pull request by MykolaAndrushchack**

### CODE REVIEW

**Comment by MykolaAndrushchack** [comment]
auth pages

File: src/app/auth/sign-up/sign-up.component.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            minlength: 'Password must have minimum 8 symbols',
+        },
+        passwordConfirm: {
+            required: 'Confirm password is required',
+            mustMatch: 'This field must match with Password',
+        },
+    };
+
+    constructor(private formBuilder: FormBuilder) {
+        super();
```
Comment (ravlik) [inline src/app/auth/sign-up/sign-up.component.ts]: try use `inject(Form builder)`

File: src/app/auth/sign-up/sign-up.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+                <input matInput type="password" placeholder="Password" name="password" formControlName="password">
+            </mat-form-field>
+            <mat-error>{{errors.password}}</mat-error>
+            <mat-form-field class="input-field">
+                <input matInput type="password" placeholder="Confirm password" name="passwordConfirm" formControlName="passwordConfirm">
+            </mat-form-field>
+            <mat-error>{{errors.passwordConfirm}}</mat-error>
+            <div class="navigation">
+                <button>Sign un</button>
+                <span routerLink="/auth/sign-in" class="link-navigation">Login</span>
```
Comment (ravlik) [inline src/app/auth/sign-up/sign-up.component.html]: use `a` and remove `navigation` from class

File: src/app/auth/sign-in/sign-in.component.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        username: {
+            required: 'Username is required',
+        },
+        password: {
+            required: 'Password is required',
+            minlength: 'Password must have minimum 8 symbols',
+        },
+    };
+
+    constructor() {
```
Comment (ravlik) [inline src/app/auth/sign-in/sign-in.component.ts]: remove

File: src/app/auth/sign-in/sign-in.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        super();
+    }
+
+    protected createForm(): FormGroup {
+        return new FormGroup({
```
Comment (ravlik) [inline src/app/auth/sign-in/sign-in.component.ts]: use from builder or classes

File: src/app/auth/sign-in/sign-in.component.scss [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+.input-field{
+    margin-bottom: 5px;
+}
+
+.navigation{
+    display: flex;
+    flex-direction: column;
+}
+.link-navigation{
```
Comment (ravlik) [inline src/app/auth/sign-in/sign-in.component.scss]: the same scss(DRY)

File: src/app/auth/sign-in/sign-in.component.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+                <input matInput type="text" placeholder="Username" formControlName="username">
+            </mat-form-field>
+            <mat-error>{{errors.username}}</mat-error>
+            <mat-form-field class="input-field">
+                <input matInput type="password" placeholder="Password" name="password" formControlName="password">
+            </mat-form-field>
+            <mat-error>{{errors.password}}</mat-error>
+            <div class="navigation">
+                <button>Sign in</button>
+                <span routerLink="/auth/sign-up" class="link-navigation">Registration</span>
```
Comment (ravlik) [inline src/app/auth/sign-in/sign-in.component.html]: use `a` or `button` instead of `span`

File: src/app/auth/auth.component.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-auth',
+  templateUrl: './auth.component.html',
+  styleUrls: ['./auth.component.scss']
+})
+export class AuthComponent implements OnInit {
+
+  constructor() { }
```
Comment (ravlik) [inline src/app/auth/auth.component.ts]: remove component if you don`t need them

File: src/app/auth/auth-routing.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { RouterModule, Routes } from '@angular/router';
+import { NgModule } from '@angular/core';
+import { SignInComponent } from './sign-in/sign-in.component';
+import { SignUpComponent } from './sign-up/sign-up.component';
+
+const routes: Routes = [
+    {
+        path: '',
```
Comment (ravlik) [inline src/app/auth/auth-routing.ts]: why we need additional layer?

File: src/app/auth/sign-up/sign-up.component.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        password: {
+            required: 'Password is required',
+            minlength: 'Password must have minimum 8 symbols',
+        },
+        passwordConfirm: {
+            required: 'Confirm password is required',
+            mustMatch: 'This field must match with Password',
+        },
+    };
```
Comment (ravlik) [inline src/app/auth/sign-up/sign-up.component.ts]: private formBuilder =inject(FormBuilder)

File: src/app/auth/sign-up/sign-up.component.ts [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import { FormComponent } from '../form.component';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MustMatch } from '../password.validator';
+
+@Component({
+    selector: 'app-sign-up',
+    templateUrl: './sign-up.component.html',
+    styleUrls: ['../auth.scss'],
+})
+export class SignUpComponent extends FormComponent {
```
Comment (ravlik) [inline src/app/auth/sign-up/sign-up.component.ts]: rename `SignUpComponent` to `RegisterComponent`, also rename files

File: src/app/auth/sign-in/sign-in.component.ts [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { FormComponent } from '../form.component';
+
+@Component({
+    selector: 'app-sign-in',
+    templateUrl: './sign-in.component.html',
+    styleUrls: ['../auth.scss'],
+})
+export class SignInComponent extends FormComponent {
+    form: FormGroup;
```
Comment (ravlik) [inline src/app/auth/sign-in/sign-in.component.ts]: rename `SignInComponent ` to `LoginComponent`, also rename files

File: src/app/auth/sign-up/sign-up.component.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+            password: new FormControl(null, [
+                Validators.required,
+                Validators.minLength(8),
+            ]),
+            passwordConfirm: new FormControl(null, Validators.required),
+        }, { validator: MustMatch('password', 'passwordConfirm') });
+        return this.form;
+    }
+
+    ngOnInit(): void {
```
[the remark this answers is not in the document]
Comment (ravlik) [reply src/app/auth/sign-up/sign-up.component.ts]: ???

File: src/app/auth/sign-in/sign-in.component.ts [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+            required: 'Username is required',
+        },
+        password: {
+            required: 'Password is required',
+            minlength: 'Password must have minimum 8 symbols',
+        },
+    };
+
+    constructor(@Inject(FormBuilder) private formBuilder: FormBuilder) {
+        super();
```
Comment (ravlik) [inline src/app/auth/sign-in/sign-in.component.ts]: see SignUp component

File: src/app/auth/register/register.component.ts [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+            username: new FormControl('', Validators.required),
+            password: new FormControl(null, [
+                Validators.required,
+                Validators.minLength(8),
+            ]),
+            passwordConfirm: new FormControl(null, Validators.required),
+        }, { validator: MustMatch('password', 'passwordConfirm') });
+    }
+
+    ngOnInit(): void {
```
Comment (ravlik) [inline src/app/auth/register/register.component.ts]: ?????

File: src/app/auth/register/register.component.ts [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+            minlength: 'Password must have minimum 8 symbols',
+        },
+        passwordConfirm: {
+            required: 'Confirm password is required',
+            mustMatch: 'This field must match with Password',
+        },
+    };
+
+    constructor(private formBuilder: FormBuilder) {
+        super();
```
Comment (ravlik) [inline src/app/auth/register/register.component.ts]: duplicate

File: src/app/auth/login/login.component.ts [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+            username: new FormControl('', Validators.required),
+            password: new FormControl('', [
+                Validators.required,
+                Validators.minLength(8),
+            ]),
+        });
+    }
+
+    login() {
+        this._handleStatusChange(this.form);
```
Comment (ravlik) [inline src/app/auth/login/login.component.ts]: duplicate

File: src/app/auth/register/register.component.ts [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+            passwordConfirm: new FormControl(null, Validators.required),
+        }, { validator: MustMatch('password', 'passwordConfirm') });
+    }
+
+    ngOnInit(): void {
+        super.ngOnInit();
+    }
+
+    registration() {
+        this._handleStatusChange(this.form);
```
Comment (ravlik) [inline src/app/auth/register/register.component.ts]: duplicate

File: src/app/auth/form.component.ts [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+            return this.errorsMessages[key][errorProp];
+        }
+        return 'This field does not meet the requirements';
+    }
+
+    submit() {
+        this._handleStatusChange(this.form);
+
+        if (this.form.valid) {
+            this.submitRequest().subscribe(() => this._queryRedirect(),
```
Comment (ravlik) [inline src/app/auth/form.component.ts]: move `this._queryRedirect()` to the next line and rename to `handleSuccessSubmit`

File: src/app/auth/form.component.ts [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+                (e) => this.showErrorRequest(e));
```
Comment (ravlik) [inline src/app/auth/form.component.ts]: rename `showErrorRequest` to  `handleErrorSubmit `

File: src/app/auth/form.component.ts [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+        }
+    }
+
+    private _queryRedirect() {
+        const redirect = this.route.snapshot.queryParamMap.get('redirect');
+        window.location.href = (redirect || this._communicationConfig.authentication.redirect);
```
Comment (ravlik) [inline src/app/auth/form.component.ts]: check if this path exists `authentication.redirect`

File: package.json [hunk 21]
```diff
@@ -56,6 +56,7 @@
     "@ngx-translate/http-loader": "4.0.0",
     "cookie-parser": "1.4.4",
     "core-js": "3.1.3",
+    "ngx-toastr": "^10.2.0",
```
Comment (ravlik) [inline package.json]: remove `^`

File: src/app/app.module.ts [hunk 22]
```diff
@@ -14,6 +14,7 @@ import { Translate } from './translate/translate';
 import { Meta } from 'meta';
 import { ConfigModule } from 'config';
 import { AppConfig } from './app.config';
+import { NotifierModule } from './notifier/notifier.module';
```
Comment (ravlik) [inline src/app/app.module.ts]: use tsconfig.json and barrel file

**Pull request merged**